syntax = "proto2";

import "nanopb.proto";

enum MyEnum1 {
  ENTRY_FIRST   = 0;
  ENTRY_Second  = 1;
  EnumThird     = 3;
}

enum MY_ENUM2 {
  ENUM2_ENTRY = 0;
}

message Message1 {
  optional int32 test_value = 1;
}

message Message2 {
  optional int32 test_value = 1;
}

message MainMessage {
  optional int32    LUCKY_number  = 1;
  optional MyEnum1  MyEnum        = 2;
  optional MY_ENUM2 MyEnum2       = 3;
  optional Message1 test_message  = 4;
  optional bytes    bytes         = 5;
  repeated fixed64  values        = 6 [(nanopb).max_count = 4, (nanopb).fixed_count = true];

  oneof oneOfName {
    Message1 msg1 = 7;
    Message2 testMessage2 = 8;
  }

  message Message3 {
    optional int32 v = 9;
  }

  optional Message3 testMessage3 = 10;
}
